easybuild/framework/easyconfig/tweak.py

```diff
@@ -842,16 +973,103 @@ def map_easyconfig_to_target_tc_hierarchy(ec_spec, toolchain_mapping, targetdir=
+                    _log.info("Depending on your configuration, this will be resolved with one of the following "
+                              "easyconfigs: %s", '\n'.join(cand['path'] for cand in potential_version_matches
+                                                           if cand['version'] == highest_version))
+                    orig_dep['version'] = highest_version
+                    if orig_dep['versionsuffix'] in versonsuffix_mapping:
+                        dep['versionsuffix'] = versonsuffix_mapping[orig_dep['versionsuffix']]
+                        orig_dep['versionsuffix'] = versonsuffix_mapping[orig_dep['versionsuffix']]
+                    dep_changed = True
+
+            if dep_changed:
                 orig_dep['short_mod_name'] = ActiveMNS().det_short_module_name(dep)
                 orig_dep['full_mod_name'] = ActiveMNS().det_full_module_name(dep)
+
     # Determine the name of the modified easyconfig and dump it to target_dir
+    if parsed_ec['ec']['versionsuffix'] in versonsuffix_mapping:
+        parsed_ec['ec']['versionsuffix'] = versonsuffix_mapping[parsed_ec['ec']['versionsuffix']]
     ec_filename = '%s-%s.eb' % (parsed_ec['ec']['name'], det_full_ec_version(parsed_ec['ec']))
     tweaked_spec = os.path.join(targetdir or tempfile.gettempdir(), ec_filename)
 
     parsed_ec['ec'].dump(tweaked_spec, always_overwrite=False, backup=True)
     _log.debug("Dumped easyconfig tweaked via --try-toolchain* to %s", tweaked_spec)
 
     return tweaked_spec
+
+
+def find_potential_version_mappings(dep, toolchain_mapping, versonsuffix_mapping):
+    """
+    Find potential version mapping for a dependency in a new hierarchy
+    :param dep: dependency
+    :param toolchain_mapping: toolchain mapping used for search
+    :param versonsuffix_mapping: mapping of version suffixes (required by software with a special version suffix, such
+    as python packages)
+    :return: list of dependencies that match
+    """
+
+    potential_version_matches = []
+
+    # Find the target toolchain and create the hierarchy to search within
+    dep_tc_name = dep['toolchain']['name']
+    if dep_tc_name in toolchain_mapping:
+        search_toolchain = toolchain_mapping[dep_tc_name]
+    else:
+        # dummy
+        search_toolchain = dep['toolchain']
+    toolchain_hierarchy = get_toolchain_hierarchy(search_toolchain)
+    # Figure out what precedes the version
+    versionprefix = dep.get('versionprefix', '')
+    prefix_to_version = ''.join([x for x in [dep['name'] + '-', versionprefix] if x])
+    # Figure out the main versionsuffix (altered depending on toolchain in the loop below)
+    versionsuffix = dep.get('versionsuffix', '')
+    # If versionsuffix is in our mapping then we expect it to be updated
+    if versionsuffix in versonsuffix_mapping:
+        versionsuffix = versonsuffix_mapping[versionsuffix]
+    for toolchain in toolchain_hierarchy:
+        candidate_ver = '.*'  # using regex for *
```

Comment:
@boegel This is probably the biggest remaining issue. This needs to be configurable somehow as at present it will, for example, upgrade from Python2 to Python3. Do I restrict it to minor versions only? I could do 2 searches, one global and one with only a minor version update, if the global returns a value I use that, otherwise I use the minor version update

Reply:
Or we could maintain a blacklist for major updates?

Reply:
I've gone with prioritising minor version updates and only if that fails trying major versions, works fine for a Python dep